packages/@react-aria/test-utils/src/types.ts

```diff
@@ -0,0 +1,133 @@
+  root: HTMLElement
+}
+
+export interface ComboBoxTesterOpts extends BaseTesterOpts {
+  /**
+   * The base element for the combobox. If provided the wrapping element around the target combobox (as is the the case with a ref provided to RSP ComboBox),
+   * will automatically search for the combobox element within.
+   */
+  root: HTMLElement,
+  /**
+   * The node of the combobox trigger button if any. If not provided, we will try to automatically use any button
+   * within the `root` provided or that the `root` serves as the trigger.
+   */
+  trigger?: HTMLElement
+}
+
+export interface GridListTesterOpts extends BaseTesterOpts {}
+
+export interface ListBoxTesterOpts extends BaseTesterOpts {
+  /**
+   * A function used by the test utils to advance timers during interactions.
+   */
+  advanceTimer?: UserOpts['advanceTimer']
+}
+
+export interface MenuTesterOpts extends BaseTesterOpts {
+  /**
+   * The trigger element for the menu.
+   */
+  root: HTMLElement,
+  /**
+   * Whether the current menu is a submenu.
+   */
+  isSubmenu?: boolean
+}
+
+export interface SelectTesterOpts extends BaseTesterOpts {
+  /**
+   * The trigger element for the select. If provided the wrapping element around the target select (as is the case with a ref provided to RSP Select),
+   * will automatically search for the select's trigger element within.
+   */
+  root: HTMLElement
+}
+
+export interface TableTesterOpts extends BaseTesterOpts {
+  /**
+   * A function used by the test utils to advance timers during interactions.
+   */
+  advanceTimer?: UserOpts['advanceTimer']
+}
+
+export interface TabsTesterOpts extends BaseTesterOpts {
+  /**
+   * The horizontal layout direction, typically affected by locale.
+   * @default 'ltr'
+   */
+  direction?: Direction
+}
+
+export interface TreeTesterOpts extends BaseTesterOpts {
+  /**
+   * A function used by the test utils to advance timers during interactions.
+   */
+  advanceTimer?: UserOpts['advanceTimer']
+}
+
+export interface BaseGridRowInteractionOpts {
+  /**
+   * The index, text, or node of the row to target.
+   */
+  row: number | string | HTMLElement,
+  /**
+   * What interaction type to use when interacting with the row. Defaults to the interaction type set on the tester.
+   */
+  interactionType?: UserOpts['interactionType']
+}
+
+export interface ToggleGridRowOpts extends BaseGridRowInteractionOpts {
+  /**
+   * Whether the row needs to be long pressed to be selected. Depends on the components implementation.
+   */
+  needsLongPress?: boolean,
+  /**
+   * Whether the checkbox should be used to select the row. If false, will attempt to select the row via press.
+   * @default 'true'
+   */
+  checkboxSelection?: boolean
+}
+
+export interface GridRowActionOpts extends BaseGridRowInteractionOpts {
+  /**
+   * Whether or not the row needs a double click to trigger the row action. Depends on the components implementation.
```

Comment:
can we check for this in any way?
such as the presence of the checkbox or `aria-selected` attribute?

if aria-selected is there (false or true) but there's no checkbox, then we're in highlight

if there is no aria-selected at all, then it's not selectable?

Reply:
potentially, but it also assumes that a user is handling highlight selection visually in the same way as we are in RSP (hiding the checkboxes). I don't really see why they would differ in implementation in reality but playing it safe here